lumigator/backend/backend/services/workflows.py

```diff
@@ -87,11 +90,18 @@ async def _run_inference_eval_pipeline(
         self._tracking_client.update_workflow_status(workflow.id, WorkflowStatus.RUNNING)
 
         # wait for the inference job to complete
-        status = await self._job_service.wait_for_job_complete(inference_job.id, max_wait_time_sec=60 * 10)
+        status = await self._job_service.wait_for_job_complete(
+            inference_job.id, max_wait_time_sec=request.job_timeout_sec
+        )
         if status != JobStatus.SUCCEEDED:
             loguru.logger.error(f"Inference job {inference_job.id} failed")
+            try:
+                self._job_service._stop_job(inference_job.id)
+            except JobUpstreamError:
+                loguru.logger.error(f"Failed to stop infer job {inference_job.id}, continuing")
             self._tracking_client.update_workflow_status(workflow.id, WorkflowStatus.FAILED)
-            raise Exception(f"Inference job {inference_job.id} failed")
+            # raise Exception(f"Inference job {inference_job.id} failed")
+            return
```

Comment:
Is this commented out by accident?

Reply:
No. AFAICT the state is reported (as failed), and the requester can act on the state. There's no need to raise an exception at this point.